Fakebook.Profile/Fakebook.Profile.UnitTests/ApiTests/ProfileControllerTests.cs

```diff
@@ -217,5 +218,44 @@ public async Task CreateProfileReturnsErrorForInvalidData()
             mockedProfileRepository.Verify(x => x.CreateProfileAsync(It.IsAny<DomainProfile>()), Times.Never);
         }
         #endregion
+
+        [Fact]
+        public async Task AddFollowerAsync()
+        {
+            using var contextFactory = new ContextFactory();
+            using var context = contextFactory.CreateContext();
+
+            var user = new ClaimsPrincipal(new ClaimsIdentity(new Claim[]
+            {
+                new Claim(ClaimTypes.Name, "john werner"),
+                new Claim(ClaimTypes.NameIdentifier, "john.werner@revature.net"),
+                new Claim("custom-claim", "example claim value"),
+            }, "mock"));
+
+            var repo = new ProfileRepository(context);
+            Mock<IStorageService> mockedStorageService = new();
+            ProfileController controller = new(
+                repo,
+                mockedStorageService.Object,
+                new NullLogger<ProfileController>()
+            );
+            controller.ControllerContext = new ControllerContext()
+            {
+                HttpContext = new DefaultHttpContext() { User = user }
+            };
+
+            DomainProfile test = new DomainProfile("tdunbar@google.com", "Trevor", "Dunbar");
+            test.BirthDate = new DateTime(1994, 6, 30, 0, 0, 0, 0, DateTimeKind.Unspecified);
+
+            ProfileApiModel testProfileModel = new ProfileApiModel(test);
+
+            await repo.CreateProfileAsync(test);
+
+            var result = controller.Follow(testProfileModel);
+            Assert.NotNull(result);
+            var actionResult = Assert.IsType<Task<ActionResult>>(result);
+            var okObjectResult = Assert.IsType<OkResult>(actionResult.Result);
+
```

Comment:
- the notnull assert is redundant with these istype asserts
- these istype asserts should be isassignableto
- that actionresult is a task, so the variable name is bad.
- write async tests

write it this way instead

```
var result = await controller.Follow(testProfileModel);
Assert.IsAssignableTo<OkResult>(result);
```

Reply:
I'm getting intellisense error on IsAssignableTo, is it IsAssignableFrom?

Reply:
yeah thats what i meant

Reply:
Changed to IsAssignableFrom and removed redundant asserts. Also made the test async.